go/processor/floria/calls_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package floria
+
+import (
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+	"go.uber.org/mock/gomock"
+)
+
+func TestCalls_InterpreterResultIsHandledCorrectly(t *testing.T) {
+	tests := map[string]struct {
+		setup   func(interpreter *tosca.MockInterpreter)
+		success bool
+		output  []byte
+	}{
+		"successful": {
+			setup: func(interpreter *tosca.MockInterpreter) {
+				interpreter.EXPECT().Run(gomock.Any()).Return(tosca.Result{Success: true}, nil)
+			},
+			success: true,
+		},
+		"failed": {
+			setup: func(interpreter *tosca.MockInterpreter) {
+				interpreter.EXPECT().Run(gomock.Any()).Return(tosca.Result{Success: false}, nil)
+			},
+			success: false,
+		},
+		"output": {
+			setup: func(interpreter *tosca.MockInterpreter) {
+				interpreter.EXPECT().Run(gomock.Any()).Return(tosca.Result{Success: true, Output: []byte("some output")}, nil)
+			},
+			success: true,
+			output:  []byte("some output"),
+		},
+	}
+
+	transaction := tosca.Transaction{
+		Sender:    tosca.Address{1},
+		Recipient: &tosca.Address{2},
+	}
+
+	ctrl := gomock.NewController(t)
+	context := tosca.NewMockTransactionContext(ctrl)
+	interpreter := tosca.NewMockInterpreter(ctrl)
+
+	senderBalance := tosca.NewValue(0)
```

Comment:
shouldn't this be a non zero value ?

Reply:
These tests to not stress the value transfer and gas has already been purchased when the `call` function is executed, therefore this value should not have an impact on this test. This is explicitly shown by setting the value to 0.